subcommands/keys/tuf_updates_apply.go

```diff
@@ -34,6 +35,13 @@ func doTufUpdatesApply(cmd *cobra.Command, args []string) {
 		msg := "Failed to apply staged TUF root updates:\n%w\n"
 		var isNonFatal bool
 		if herr := client.AsHttpError(err); herr != nil {
+			if herr.Response.StatusCode == 404 {
+				// Double check: if there are no TUF updates - fail clean; otherwise, fatal error.
+				updates, err1 := api.TufRootUpdatesGet(factory)
+				if err1 == nil && updates.Status == client.TufRootUpdatesStatusNone {
+					subcommands.DieNotNil(errors.New("There are no TUF root updates in progress."))
+				}
+			}
```

Comment:
If `err1` is not nil then the command will print 
```
This is a critical error: Staged TUF root updates may be partially applied to your Factory.
Please, re-run the "fioctl keys tuf updates apply" again after some time, but as soon as possible.
If the error persists, please, contact the customer support.`
```
It looks like too scary message for 404.

Reply:
If we get 404 - we check if the updates transaction exist:
- if it doesn't - that's regular situation, and we show a simple regular error.
- if it exists, and there was 404 - only the universe knows why it happened, so it is critical.

Reply:
The `api.TufRootUpdatesGet(factory)` can fail so `err1` is not nil if there is no networking connection e.g. And the command will print that scary message then, instead of connection error?

Reply:
There are 3 use cases:
- `err == nil & there is no transtaction` - we know why 404 happened, and this is a regular situation.
- `err == nil & there is transtaction` - we don't know why 404 happened, this is bad. So, we show critical error.
- `err != nil` - we don't know what we don't know, this is the worst situation possible. So, we show critical error _with the original error message_.

Reply:
If `err1 != nil` we know the exact error, it's in the `err1` context. As I mentioned in the example above I would prefer to see "connection/timeout error" instead of that scary message.

Anyway, it's not really important.

Reply:
> As I mentioned in the example above I would prefer to see "connection/timeout error" instead of that scary message.

Here is the thing:
- we apply updates and get 404 error;
- we try to verify if it is critical or not... for that we see if the transaction exists;
- so, checking for transaction is a tool to evaluate the error severity; it has no other value beyond that.
- therefore, if we get an error while ~checking if transaction exists~ assessing the error severity - that severity gets a maximum level.
- ...because we know that something bad has happened, and we cannot assess what exactly.